src/tribler/core/upgrade/tests/test_version_manager.py

```diff
@@ -205,6 +230,47 @@ def test_fork_state_directory(tmpdir_factory):
     assert history2.last_run_version.version_str == code_version_id
 
 
+disk_space_testdata = [
+    (100, 99, True),
+    (100, 100, True),
+    (100, 101, False),
+]
+
+
+@pytest.mark.parametrize("space_required, space_available, expected_exception", disk_space_testdata)
+def test_fork_state_directory_with_storage_constraints_param(space_required, space_available, expected_exception,
+                                                             version_history):
+    version_history.code_version.can_be_copied_from.upgrade_size = lambda: space_required
+    version_history.free_disk_space = lambda: space_available
+
+    try:
+        _ = version_history.fork_state_directory_if_necessary()
+    except NoDiskSpaceAvailableError:
+        if not expected_exception:
```

Comment:
**nitpick:** it should be possible to use `with pytest.raises(NoDiskSpaceAvailableError): ...` here